src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -295,18 +305,24 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
   void MarkDebugMarkerEnd(VkCommandBuffer command_buffer) {
     absl::WriterMutexLock lock(&mutex_);
     bool marker_depth_exceeds_maximum;
-    {
-      CHECK(command_buffer_to_state_.contains(command_buffer));
-      CommandBufferState& state = command_buffer_to_state_.at(command_buffer);
-      marker_depth_exceeds_maximum =
-          state.local_marker_stack_size > max_local_marker_depth_per_command_buffer_;
-      Marker marker{.type = MarkerType::kDebugMarkerEnd, .cut_off = marker_depth_exceeds_maximum};
-      state.markers.emplace_back(std::move(marker));
-      // We might see more "ends" than "begins", as the "begins" can be on a different command
-      // buffer.
-      if (state.local_marker_stack_size != 0) {
-        --state.local_marker_stack_size;
-      }
+
+    if (!command_buffer_to_state_.contains(command_buffer)) {
+      ERROR_ONCE(
+          "Calling vkCmdDebugMarkerEndEXT/vkCmdEndDebugUtilsLabelEXT on a command buffer "
+          "that is in the initial state (i.e. either freshly allocated or reset with "
+          "vkResetCommandBuffer).");
+      return;
+    }
+    CHECK(command_buffer_to_state_.contains(command_buffer));
+    CommandBufferState& state = command_buffer_to_state_.at(command_buffer);
+    marker_depth_exceeds_maximum =
+        state.local_marker_stack_size > max_local_marker_depth_per_command_buffer_;
+    Marker marker{.type = MarkerType::kDebugMarkerEnd, .cut_off = marker_depth_exceeds_maximum};
+    state.markers.emplace_back(std::move(marker));
+    // We might see more "ends" than "begins", as the "begins" can be on a different command
+    // buffer.
+    if (state.local_marker_stack_size != 0) {
```

Comment:
Nit: I find it slightly more readable if you say > 0 here. (But since local_marker_stack_size is unsigned, this doesn't actually make a different.)

Reply:
Agree. Done.